opendps/settings_calibration.c

```diff
@@ -0,0 +1,375 @@
+    .get_parameter = &get_parameter,
+    .num_items = 5,
+    .parameters = {
+        {
+            .name = "V_DAC",
+            .unit = unit_none,
+            .prefix = si_none
+        },
+        {
+            .name = "A_DAC",
+            .unit = unit_none,
+            .prefix = si_none
+        },
+        {
+            .name = {'\0'} /** Terminator */
+        },
+    },
+    .items = { (ui_item_t*) &calibration_v_dac,
+               (ui_item_t*) &calibration_a_dac,
+               (ui_item_t*) &calibration_vin_adc,
+               (ui_item_t*) &calibration_v_adc,
+               (ui_item_t*) &calibration_a_adc }
+};
+
+/**
+ * @brief      Set function parameter
+ *
+ * @param[in]  name   name of parameter
+ * @param[in]  value  value of parameter as a string - always in SI units
+ *
+ * @retval     set_param_status_t status code
+ */
+static set_param_status_t set_parameter(char *name, char *value)
+{
+    int32_t ivalue = atoi(value);
+    if (strcmp("V_DAC", name) == 0) {
+        if (ivalue < calibration_v_dac.min || ivalue > calibration_v_dac.max) {
+            emu_printf("[Calibration] V_DAC %d is out of range (min:%d max:%d)\n", ivalue, calibration_v_dac.min, calibration_v_dac.max);
+            return ps_range_error;
+        }
+        emu_printf("[Calibration] Setting V_DAC to %d\n", ivalue);
+        calibration_v_dac.value = ivalue;
+        v_dac_changed(&calibration_v_dac);
+        return ps_ok;
+    } else if (strcmp("A_DAC", name) == 0) {
+        if (ivalue < calibration_a_dac.min || ivalue > calibration_a_dac.max) {
+            emu_printf("[Calibration] A_DAC %d is out of range (min:%d max:%d)\n", ivalue, calibration_a_dac.min, calibration_a_dac.max);
+            return ps_range_error;
+        }
+        emu_printf("[Calibration] Setting A_DAC to %d\n", ivalue);
+        calibration_a_dac.value = ivalue;
+        a_dac_changed(&calibration_a_dac);
+        return ps_ok;
+    }
+    return ps_unknown_name;
+}
+
+/**
+ * @brief      Get function parameter
+ *
+ * @param[in]  name       name of parameter
+ * @param[in]  value      value of parameter as a string - always in SI units
+ * @param[in]  value_len  length of value buffer
+ *
+ * @retval     set_param_status_t status code
+ */
+static set_param_status_t get_parameter(char *name, char *value, uint32_t value_len)
+{
+    if (strcmp("V_DAC", name) == 0) {
+        (void) mini_snprintf(value, value_len, "%d", calibration_v_dac.value);
+        return ps_ok;
+    } else if (strcmp("A_DAC", name) == 0) {
+        (void) mini_snprintf(value, value_len, "%d", calibration_a_dac.value);
+        return ps_ok;
+    }
+    return ps_unknown_name;
+}
+
+/**
+ * @brief      Callback for when the function is enabled
+ *
+ * @param[in]  enabled  true when function is enabled
+ */
+static void calibration_enable(bool enabled)
+{
+    emu_printf("[Calibration] %s output\n", enabled ? "Enable" : "Disable");
+    if (enabled) {
+        pwrctl_set_ilimit(10000);
```

Comment:
Is this magic number intended? ;)

Reply:
Ooops, This it to stop the OCP firing. I've added a comment to it to clarify